packages/cli/src/lib/plugins/fingerprint.ts

```diff
@@ -17,28 +20,172 @@ type NativeFingerprintCommandOptions = {
   raw?: boolean;
 };
 
+type EnvFingerprintSource = {
+  type: 'env';
+  name: string;
+  value: string | undefined;
+  hash: string | null;
+};
+
+type EnvConfig = {
+  common?: string[];
+  ios?: string[];
+  android?: string[];
+};
+
+const DEFAULT_ENV_VARS: EnvConfig = {
+  ios: [
+    'RCT_NEW_ARCH_ENABLED',
+    'USE_HERMES',
+    'USE_FRAMEWORKS',
+    'RCT_ENABLE_PREBUILD',
+  ],
+  android: [
+    'ORG_GRADLE_PROJECT_newArchEnabled',
+    'ORG_GRADLE_PROJECT_hermesEnabled',
+    'REACT_NATIVE_OVERRIDE_VERSION',
+  ],
+  common: [
+    'NODE_ENV',
+    'CI',
```

Comment:
I don't think we want to take CI into consideration, because it will result in different fingerprints for local vs CI builds